fix(transformer): only run typescript plugin for typescript source

Boshen opened the pull request:
closes #6865

TypeScript plugin changes import / export statements so it needs to be turned off for non-typescript files. This should also give a little performance boost for non-typescript files.

Boshen commented:
* **#6889** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/6889?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Boshen and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

overlookmotel commented:
> ⚡	`transformer[RadixUIAdoptionSection.jsx]`	157.6 µs	152.8 µs	+3.1%
> ❌	`transformer[checker.ts]`	18.7 ms	19.3 ms	-3.18%

I assume the regression for `checker.ts` is the cost of the extra "is TS enabled" checks. We may be able to make those checks cheaper by using an `is_ts_enabled` boolean flag instead. But, as you said in https://github.com/oxc-project/oxc/issues/6865#issuecomment-2436626895, we also want to avoid constructing the TS transformer if it's disabled - that setup cost will be significant for small JS files.

We should probably look at how transforms are constructed, and how options are stored, for all the transforms together once we've completed the rest of the ES6 transforms. We can probably gain some perf by refactoring it.

overlookmotel reviewed (approved):
> ⚡	`transformer[RadixUIAdoptionSection.jsx]`	157.6 µs	152.8 µs	+3.1%
> ❌	`transformer[checker.ts]`	18.7 ms	19.3 ms	-3.18%

I assume the regression for `checker.ts` is the cost of the extra "is TS enabled?" checks. We may be able to make those checks cheaper by using an `is_ts_enabled` boolean flag instead. But, as you said in https://github.com/oxc-project/oxc/issues/6865#issuecomment-2436626895, we also want to avoid constructing the TS transformer if it's disabled - that setup cost will be significant for small JS files.

We should probably look at how transforms are constructed, and how options are stored, for all the transforms together once we've completed the rest of the ES6 transforms. We can probably gain some perf by refactoring it, one way or another.

But for now I say merge it! It speeds up all the benchmarks except `checker.ts`, and fixes loads of tests.

Boshen commented:
### Merge activity

* **Oct 25, 9:59 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Boshen committed:
fix(transformer): only run typescript plugin for typescript source (#6889)

closes #6865

TypeScript plugin changes import / export statements so it needs to be turned off for non-typescript files. This should also give a little performance boost for non-typescript files.
